### PR TITLE
Eureka instructions

### PR DESCRIPTION
**Pull request by ijsnellf**
1st draft of Eureka instructions. There is quite a bit of duplication between the Consul and Eureka instructions right now.

### CODE REVIEW

**Comment by ijsnellf** [comment]
fixes #591

File: _docs/guides/bookinfo.md [hunk 1]
```diff
@@ -174,23 +174,28 @@ To start the application, follow the instructions below corresponding to your Is
    export GATEWAY_URL=$(kubectl get po -n istio-system -l istio=ingress -o 'jsonpath={.items[0].status.hostIP}'):$(kubectl get svc istio-ingress -n istio-system -o 'jsonpath={.spec.ports[0].nodePort}')
    ```
 
-### Running on Docker with Consul
+### Running on Docker
 
 1. Change directory to the root of the Istio installation directory.
```
Comment (rshriram) [inline _docs/guides/bookinfo.md]: Add a line saying there are two options: Consul and Eureka.. blah blah

File: _docs/guides/bookinfo.md [hunk 2]
```diff
@@ -174,23 +174,28 @@ To start the application, follow the instructions below corresponding to your Is
...
 
-1. Bring up the Istio control plane and the application containers:
+1. Bring up the application containers.
 
-    ```bash
-    docker-compose -f samples/bookinfo/consul/bookinfo.yaml up -d
-    ```
+    1. For Consul run:
```
Comment (rshriram) [inline _docs/guides/bookinfo.md]: To test with Consul, run the following command:

File: _docs/guides/bookinfo.md [hunk 3]
```diff
@@ -174,23 +174,28 @@ To start the application, follow the instructions below corresponding to your Is
...
+        ```bash
+        docker-compose -f samples/bookinfo/consul/bookinfo.yaml up -d
+        ```
+    1. For Eureka run:
```
Comment (rshriram) [inline _docs/guides/bookinfo.md]: same as above

File: _docs/setup/eureka/index.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+title: Eureka
+overview: Instructions for installing the Istio control plane in a Eureka based environment.
+
+order: 20
```
Comment (rshriram) [inline _docs/setup/eureka/index.md]: are you sure this is the right order? Add +10 after Consul/index.md and then change the order value in index.md for CloudFoundry, Mesos.

File: _docs/setup/eureka/install.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+
+### Other Istio Components
+
+Debian packages for Istio Pilot, Mixer, and CA are available through the
+Istio release. Alternatively, these components can be run as Docker
+containers (docker.io/istio/pilot, docker.io/istio/mixer,
+docker.io/istio/istio-ca). Note that these components are stateless and can
+be scaled horizontally. Each of these components depends on the Istio API
+server, which in turn depends on the etcd cluster for persistence. To
```
Comment (rshriram) [inline _docs/setup/eureka/install.md]: This doesn't belong in Eureka. No Nomad here

File: _docs/setup/eureka/install.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+## Adding Sidecars to Service Instances
+
+Each instance of a service in an application must be accompanied by the
+Istio sidecar. Depending on the unit of your installation (Docker
+containers, VM, bare metal nodes), the Istio sidecar needs to be installed
+into these components.  For example, if your infrastructure uses VMs, the
+Istio sidecar process must be run on each VM that needs to be part of the
+service mesh.
+
+One way to package the sidecars into a Nomad-based deployment is to add the
```
Comment (rshriram) [inline _docs/setup/eureka/install.md]: Same here as well.

File: _docs/setup/eureka/install.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+the Istio sidecars. The IP table script to setup such forwarding can be
+found in the
+[Istio Pilot repository](https://github.com/istio/pilot/blob/master/docker/prepare_proxy.sh). 
+
+> Note: This script must be executed before starting the application or
+> the sidecar process. 
+
+Refer to the
+[sidecar installation](https://github.com/istio/istio/blob/master/samples/bookinfo/consul/build-docker-services.sh)
+for Bookinfo services to understand the steps needed to achieve transparent proxying.
```
Comment (rshriram) [inline _docs/setup/eureka/install.md]: May be axe this entire doc

File: _docs/setup/eureka/index.md (hunk 4 above)
Comment (ijsnellf) [reply _docs/setup/eureka/index.md]: nice catch

File: _docs/setup/eureka/install.md (hunk 7 above)
Comment (ijsnellf) [reply _docs/setup/eureka/install.md]: all of `_docs/setup/eureka/install.md`?

**Comment by ijsnellf** [comment]
hold off on merging this --I need to verify everything still works with the zipkin/tracing changes.

**Comment by rshriram** [comment]
Yes, especially with Zipkin. I did test Zipkin with Consul.

**Comment by ijsnellf** [comment]
Zipkin is not part of the istio/istio docker-compose file for Eureka right now. But I verified it still works nevertheless.

I'll create a PR for Zipkin in Eureka in istio/istio, but the docs don't need to be gated on it.

**Pull request merged**